addons/mail/static/src/components/composer_drop_list_suggestion/composer_drop_list_suggestion.js

```diff
@@ -31,8 +31,16 @@ class PartnerMentionSuggestion extends Component {
     /**
      * @returns {mail.partner}
      */
-    get partner() {
-        return this.env.models['mail.partner'].get(this.props.partnerLocalId);
+    get model() {
+        return this.env.models[this.props.modelName].get(this.props.elementLocalId);
+    }
+
+    get isCannedResponse() {
+        return this.props.modelName === "mail.canned_response";
+    }
+
+    get isPartner() {
+        return this.props.modelName === "mail.partner";
```

Comment:
Those are very simple getters used only once, so avoid indirection and put the conditions directly in the template